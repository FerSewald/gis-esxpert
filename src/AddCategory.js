import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setIInputValue] = useState('')
    const handleInputChange = (e) => {
        
        setIInputValue(e.target.value)
    }
    const handelSubmit   = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats]);
            setIInputValue('');
        }

    }

    return (
        <form onSubmit={handelSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
}