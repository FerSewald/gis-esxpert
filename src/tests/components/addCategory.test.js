import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { AddCategory } from '../../AddCategory';

describe('Pruevas en <AddCategory />', () => {
    const setCategories = () => {};

    const wrapper = shallow(<AddCategory setCategories={setCategories}/>)



    test('debe de mostrase correctamete', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('deve cambiar el input', () => {
        
        const input = wrapper.find('input');
        const value = 'hola mundo' 

        input.simulate('change', {target: { value: value } });

        expect(wrapper.find('p').text().trim()).toBe(value);

    })
    
    
});
