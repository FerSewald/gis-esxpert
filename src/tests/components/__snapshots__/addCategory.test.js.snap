// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en <AddCategory /> debe de mostrase correctamete 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    <Component />
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
